compiler/rustc_macros/src/query.rs

```diff
@@ -272,6 +272,14 @@ fn process_modifiers(query: &mut Query) -> QueryModifiers {
                 if desc.is_some() {
                     panic!("duplicate modifier `desc` for query `{}`", query.name);
                 }
+                if query.doc_comments.is_empty() {
+                    // FIXME: replace `{}` with `_` somehow
+                    let format_str = list.first().unwrap();
```

Comment:
I think you can match on it and have a match arm for the `Expr::Lit(expr_lit)` case, then match on `expr_lit.lit` to get `Lit::Str(lit_str)` then call `lit_str.value()` to get its value as a `String`. Then do `string.replace("{}", "_")` and change the code in the `parse_quote!` invocation to `#[doc = "#str"]`. I *think* that might work!

Reply:
Not sure if you want to panic if it's not an `Expr::Lit(ExprLit::Str(lit_str))` or just leave the node as-is. Panicking is probably fine since it would happen at compile-time (thus it shouldn't be able to cause ICEs).